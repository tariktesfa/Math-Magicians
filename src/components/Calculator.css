.container {
  display: flex;
  justify-content: space-between;
}

.text {
  padding: 55px;
}

.calculator {
  position: absolute;
  background-color: #a1a1a1;
  top: 20%;
  left: 40%;
  height: 284px;
  width: 300px;
  border: none;
}

.display {
  text-align: right;
  font-size: 24px;
  padding: 10px;
  color: #fff;
  border: 1px #d3d3d3;
}

.button-keys {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.button {
  color: #000;
  font-size: 16px;
  padding: 0;
  border: 1px solid #d3d3d3;
  width: 75px;
  height: 47px;
}

.button:nth-child(17) {
  grid-column: 1/span 2;
  width: 150px;
}

.button:nth-child(4),
.button:nth-child(8),
.button:nth-child(12),
.button:nth-child(16),
.button:nth-child(19) {
  background-color: #f09845;
}
