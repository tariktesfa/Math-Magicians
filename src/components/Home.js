import React from 'react';

const Home = () => (
  <section className="home" data-testid="home-page">
    <h4>Welcome to Math Magicians!</h4>
    <br />
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Voluptates eos
      deleniti, ex amet adipisci eveniet, optio reiciendis velit iusto fugiat
      placeat sunt, omnis error atque. Quis officiis sit ad ratione.
    </p>
    <br />
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis nisi
      perferendis, tenetur tempore corrupti exercitationem velit, repellendus
      nulla quibusdam voluptatibus fugit doloribus magni eos beatae officiis cum
      aspernatur dolore autem?
    </p>
  </section>
);

export default Home;
