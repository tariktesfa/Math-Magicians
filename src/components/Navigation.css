nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 0;
}

nav ul {
  display: flex;
  gap: 2rem;
  list-style: none;
}

nav li a {
  border-bottom: 2px solid #fff;
}

nav li a:hover {
  border-color: #000;
}

nav li a.active {
  border-color: #f5913e;
}
