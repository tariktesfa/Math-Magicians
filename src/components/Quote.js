import React from 'react';

const Quote = () => {
  const list = [
    'Strength does not come from physical capacity. It comes from an indomitable will.',
    'Nothing is impossible, the word itself says "I\'m possible"!',
    'The secret of getting ahead is getting started.',
    "I've missed more than 9,000 shots during my career. I've lost almost 300 games. 26 times, I've been trusted to take the game winning shot and missed. I've failed over and over and over again in my life. And that is why I succeed.",
    'The greatest enemy of knowledge is not ignorance, it is the illusion of knowledge.',
    'By seeking and blundering we learn.',
    'The price of anything is the amount of life you exchange for it.',
    'A change in perspective is worth 80 IQ points.',
  ];

  return (
    <section className="quote">
      <h2 data-testid="quote">{list[Math.floor(Math.random() * 8)]}</h2>
    </section>
  );
};

export default Quote;
