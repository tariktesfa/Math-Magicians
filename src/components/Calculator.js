import React, { useState } from 'react';
import './Calculator.css';
import Calculate from '../logic/Calculate';

const Calculator = () => {
  const [data, setData] = useState({
    currentValue: 0,
    operator: '',
    nextValue: '',
  });

  const handleClick = (e) => {
    const result = Calculate(data, e.target.textContent);
    setData(result);
  };

  const display = `${data.currentValue || data.operator || data.nextValue ? `${data.currentValue || ''} ${data.operator || ''} ${data.nextValue || ''}` : '0'} `;
  const keyArray = ['AC', '+/-', '%', 'รท', 7, 8, 9, 'x', 4, 5, 6, '-', 1, 2, 3, '+', 0, '.', '=']
    .map((keys) => <button type="button" className="button" data-testid={keys} key={keys} id={keys} onClick={handleClick}>{keys}</button>);

  return (
    <div className="container">
      <div className="text"><h3>Let us do some Maths!!</h3></div>
      <div className="calculator">
        <div className="display" data-testid="display">{display}</div>
        <div className="button-keys">{keyArray}</div>
      </div>
    </div>
  );
};

export default Calculator;
