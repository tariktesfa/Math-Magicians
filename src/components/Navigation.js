import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navigation.css';

const Navigation = () => (
  <nav data-testid="navbar">
    <h1>Math Magicians</h1>
    <ul>
      <li>
        <NavLink
          className={({ isActive }) => (isActive ? 'active' : '')}
          to="/"
        >
          Home
        </NavLink>
      </li>
      <li>
        <NavLink
          className={({ isActive }) => (isActive ? 'active' : '')}
          to="/calculator"
        >
          Calculator
        </NavLink>
      </li>
      <li>
        <NavLink
          className={({ isActive }) => (isActive ? 'active' : '')}
          to="/quote"
        >
          Quotes
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navigation;
