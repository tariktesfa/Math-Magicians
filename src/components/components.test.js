import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import {
  render, screen, cleanup, fireEvent,
} from '@testing-library/react';

import Quote from './Quote';
import Navigation from './Navigation';
import Calculator from './Calculator';
import Home from './Home';

afterEach(() => {
  cleanup();
});

// --------------Home test------------

describe('should render the test for Home', () => {
  render(<Home />);
  const homePage = screen.getByTestId('home-page');
  expect(homePage).toBeTruthy();

  test('test render snapshot for home page', () => {
    const tree = renderer.create(<Router><Home /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// --------------Quotes page---------------

test('should render the quote component', () => {
  render(<Quote />);
  const quotes = screen.getAllByTestId('quotes');
  expect(quotes).toBeTruthy();
});

// ---------------navigation bar----------------

describe('run test to render the navigation', () => {
  test('test for nav component', () => {
    render(
      <Router>
        <Navigation />
      </Router>,
    );

    const navBar = screen.getByTestId('navbar');

    expect(navBar).toBeTruthy();
  });

  test('test navbar snapshot', () => {
    const tree = renderer.create(<Router><Navigation /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// -------------------calculator------------------

describe('run test for the calculator component', () => {
  test('run to check events to the calculator', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByTestId('7'));
    fireEvent.click(screen.getByTestId('+'));
    fireEvent.click(screen.getByTestId('9'));
    fireEvent.click(screen.getByTestId('='));
    const result = screen.getByTestId('display');
    expect(result.innerHTML).toBe('16   ');
  });

  test('test to render snapshot for calculator', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
