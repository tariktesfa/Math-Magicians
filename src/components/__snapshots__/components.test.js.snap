// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run test for the calculator component test to render snapshot for calculator 1`] = `
<div
  className="container"
>
  <div
    className="text"
  >
    <h3>
      Let us do some Maths!!
    </h3>
  </div>
  <div
    className="calculator"
  >
    <div
      className="display"
      data-testid="display"
    >
      0 
    </div>
    <div
      className="button-keys"
    >
      <button
        className="button"
        data-testid="AC"
        id="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        data-testid="+/-"
        id="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        data-testid="%"
        id="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button"
        data-testid="รท"
        id="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="button"
        data-testid={7}
        id={7}
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        data-testid={8}
        id={8}
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        data-testid={9}
        id={9}
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button"
        data-testid="x"
        id="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="button"
        data-testid={4}
        id={4}
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        data-testid={5}
        id={5}
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        data-testid={6}
        id={6}
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button"
        data-testid="-"
        id="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="button"
        data-testid={1}
        id={1}
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button"
        data-testid={2}
        id={2}
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        data-testid={3}
        id={3}
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button"
        data-testid="+"
        id="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="button"
        data-testid={0}
        id={0}
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button"
        data-testid="."
        id="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button"
        data-testid="="
        id="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`run test to render the navigation test navbar snapshot 1`] = `
<nav
  data-testid="navbar"
>
  <h1>
    Math Magicians
  </h1>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;

exports[`should render the test for Home test render snapshot for home page 1`] = `
<section
  className="home"
  data-testid="home-page"
>
  <h4>
    Welcome to Math Magicians!
  </h4>
  <br />
  <p>
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Voluptates eos deleniti, ex amet adipisci eveniet, optio reiciendis velit iusto fugiat placeat sunt, omnis error atque. Quis officiis sit ad ratione.
  </p>
  <br />
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis nisi perferendis, tenetur tempore corrupti exercitationem velit, repellendus nulla quibusdam voluptatibus fugit doloribus magni eos beatae officiis cum aspernatur dolore autem?
  </p>
</section>
`;
